pkg/controller/nodeipam/ipam/range_allocator_test.go

```diff
@@ -517,13 +560,23 @@ func TestAllocateOrOccupyCIDRSuccess(t *testing.T) {
 					t.Fatalf("%v: unexpected error when occupying CIDR %v: %v", tc.description, allocated, err)
 				}
 			}
-			if err := allocator.AllocateOrOccupyCIDR(tc.fakeNodeHandler.Existing[0]); err != nil {
-				t.Errorf("%v: unexpected error in AllocateOrOccupyCIDR: %v", tc.description, err)
+		}
+
+		updateCount := 0
+		for _, node := range tc.fakeNodeHandler.Existing {
+			if node.Spec.PodCIDRs == nil {
+				updateCount++
 			}
```

Comment:
it will work the same with the previous code, no?
why iterating now?

Reply:
I just wanted to make sure that all tests have exactly one node without assigned PodCIDRs.

Reply:
The iteration is done because we now have more than one node, and the one that needs an address is not always `Existing[0]`.